### ISSUE TITLE
Braintree: sometimes payment profile ID isn't set

### ISSUE THREAD

**Issue by nosegraze** [restrictcontentpro/restrict-content-pro#1308]
I ran into a case where my Braintree payment profile ID was never set, and as a result, the webhooks were always dying with `no user ID found`. This only happens under some specific circumstances.

- Have a payment profile ID for a different gateway set (I had one for Authorize.net set).
- This causes the `$payment_profile_id` variable to default to `bt_$this->user_id`.
- And you'll need to sign up with an account where that customer ID already exists in Braintree.
- However, this only runs `$member->set_payment_profile_id( $customer->id );` if the customer _didn't_ already exist.

So the gateway has created an ID and is able to look up the customer with it, but since the customer already exists it never saves that ID in the user profile so I still end up stuck with my Authorize.net one.

tl;dr: payment profile ID should always be set, whether the customer already exists or is being newly created.

**Comment by nosegraze** [comment]
PR: #1392
